src/components/ScholarshipForm.jsx

```diff
@@ -239,59 +242,97 @@ function ScholarshipForm({ scholarship }) {
+      <form onSubmit={formik.handleSubmit}>
+        <Stepper activeStep={activeStep} orientation="vertical">
+          {Object.entries(stepperItems).map(
+            ([label, { description, content }]) => (
+              <Step key={label}>
+                <StepLabel>{label}</StepLabel>
+                <StepContent>
+                  <Typography>{description}</Typography>
+                  <Box marginY={3}>{content}</Box>
+                  <Button
+                    disabled={activeStep === 0}
+                    onClick={() => setActiveStep((prevStep) => prevStep - 1)}>
+                    BACK
+                  </Button>
+                  <Button
+                    key={activeStep}
+                    variant="contained"
+                    color="primary"
+                    disabled={formik.isSubmitting}
+                    type={onLastStep ? 'submit' : 'button'}
+                    onClick={() => {
+                      if (onLastStep) return;
+                      formik.validateForm().then((errors) => {
+                        const checkboxError = validationCheck();
+                        if (checkboxError)
+                          errors = { ...errors, checkbox: checkboxError };
+
+                        if (Object.keys(errors).length === 0)
+                          setActiveStep((prevStep) => prevStep + 1);
 
-                      if (Object.keys(errors).length === 0)
-                        setActiveStep((prevStep) => prevStep + 1);
+                        return formik.setErrors(errors);
+                      });
+                    }}>
+                    {onLastStep ? 'Submit' : 'Next'}
+                  </Button>
+                  {submissionError && (
+                    <Alert
+                      severity="error"
+                      onClose={() => setSubmissionError(null)}>
+                      <AlertTitle>
+                        There was an error submitting your changes:
+                      </AlertTitle>
+                      {submissionError.toString()}
+                    </Alert>
+                  )}
+                </StepContent>
+              </Step>
+            )
+          )}
+        </Stepper>
+      </form>
 
-                      return formik.setErrors(errors);
-                    });
-                  }}>
-                  {onLastStep ? 'Submit' : 'Next'}
-                </Button>
-                {submissionError && (
-                  <Alert
-                    severity="error"
-                    onClose={() => setSubmissionError(null)}>
-                    <AlertTitle>
-                      There was an error submitting your changes:
-                    </AlertTitle>
-                    {submissionError.toString()}
-                  </Alert>
-                )}
-              </StepContent>
-            </Step>
-          )
-        )}
-      </Stepper>
-    </form>
+      {openAlert && Object.keys(lintIssues?.issues).length > 0 && (
```

Comment:
I was thinking of making a separate function that returns `{messages, values}` specific to requirements, that way this alert wont show when there are lint errors not related to the requirements

Reply:
Hmm, yeah there are some lint issues that aren't automatically addressable. What if you return something that looks like`{messages: string[], fixes: {requirements: {gpa: number, ...}}}`?

Or you could make `fixes` an array of `{key: string, values: any[]}` objects.